### ISSUE TITLE
Questions about the Repo

### ISSUE THREAD

**Issue by sscaff1** [ahrefs/bs-reactstrap#12]
Hello! I would like to work on and contribute to this repo. I'm using it for a couple of projects. Couple of questions:

1. Why are all of the files called BsReactstrap_{Component} instead of just {Component}? Would you be open to a PR that changes all the file names to just Component? You'll still reference it with BsReactstrap.Alert rather than BsReactstrap.BsReactstrap_Alert which might make it nicer to use.
2. Can I introduce the new object bucklescript syntax? This would just clean up the bindings and remove now depreciated syntax.
Now:
```reason
[@bs.obj]
external makeProps : (~className: string=?, unit) = _ = "";
```
After:
```reason
[@bs.deriving abstract]
type props = {
  [@bs.optional]
  className: string
};
```
then in the interface file
```reason
  let props: (~className: string=?, unit) => props;
```
3. Can I update the bs-platform version? 
4. Can I remove the in-source: true? Seeing all JS files just junks up workspace for a contributer.

Let me know if you'd be open to accepting a PR for some or all of these things. Thanks again for starting these bindings!

**Comment by liubko** [comment]
Hi!

1. You can reference files like `BsReactstrap.Alert`. Because we do provide [this file](https://github.com/ahrefs/bs-reactstrap/blob/master/src/BsReactstrap.re). We attach `BsReactstrap_` prefix to avoid possible name collisions with the files in consumer's project.

2. yes, I would like to use the new syntax. I don't think we need interface files.

3. :+1:

4. yes, I think that's might be better. @rusty-key what do you think? [Link](https://bucklescript.github.io/docs/en/build-configuration.html#package-specs) for the reference. It wouldn't change anything for reason based consumer projects, it would only change things If someone was using it from JS project, but that would be pointlles. Did I get it right?

**Comment by sscaff1** [comment]
1. I see that, but I don't understand the point of the `BsReactstrap__` prefix unless you're literally copying a pasting files. You can have the files just name alert.re and it's unnecessary to have the BsReactstrap file. You just need to turn on namespace in order to make this work. https://bucklescript.github.io/docs/en/build-configuration.html#name-namespace. Easy change to the bsconfig.json, and then BsReactstrap__Alert can turn into alert.re (lowercase seems to be more the norm - it'll still be available as Alert). Uppercase/Lowercase seems to be a matter of preference so let me know what you prefer.
2. The only reason for the interface file is that the new syntax produces bindings for every property so it looks a little junky as a consumer trying to use it. You'll see stuff like this:
`  [@bs...] external tag: props => option(tag) = "tag";` for all the properties.
4. Basically instead of seeing Alert.bs.js right next to the .re file. You would see Alert.bs.js in the lib/js folder. This is common for bindings.

I hope to get in a PR by this weekend.

**Comment by liubko** [comment]
1. it’s published under name `@ahrefs/bs-reactstrap`, so if `"namespace": true` it will be accessible as `AhrefsBsReactstrap` instead of `BsReactstrap`, bsb doesn't allow to configure the name of the namespace

**Comment by sscaff1** [comment]
I see - thanks for the clarification. There is another package called bs-reactstrap - sorry about the confusion.

**Comment by sscaff1** [comment]
Closed with #13
